webapp/channels/src/utils/markdown/renderer.tsx

```diff
@@ -136,6 +136,13 @@ export default class Renderer extends marked.Renderer {
     public link(href: string, title: string, text: string, isUrl = false) {
         let outHref = href;
 
+        if (this.formattingOptions.unsafeLinks && !href.startsWith('/')) {
```

Comment:
What about `//example.com/?stuff+goes+here`? What about `/api/v4/image?url=https://example.com/?stuff+goes+here`?

Reply:
Modified so that only URLs that start with the SiteURL and don't hit the image proxy are accepted.